wt-status.c

```diff
@@ -664,13 +664,14 @@ static int add_file_to_list(const struct object_id *oid,
 	struct string_list_item *it;
 	struct wt_status_change_data *d;
 	struct wt_status *s = context;
-	char *full_name;
+	struct strbuf full_name = STRBUF_INIT;
```

Comment:
This is called potentially a _lot_ of times. If it turns out that we need to switch between sparse and full indices longer (or more often) than anticipated, we could pass in an initialized `strbuf` and only `strbuf_reset()` it in `add_file_to_list()`.

Reply:
The string is eventually added to a `struct string_list` that doesn't duplicate the strings. Thus, we will need to allocate these strings no matter what.

This case is also rare: we need to have the index differ from `HEAD` on a sparse directory entry, which requires strange situations. A performance drop in this situation is the least of the user's worries.